var/spack/repos/builtin/packages/py-poetry/package.py

```diff
@@ -21,43 +22,59 @@ class PyPoetry(PythonPackage):
     depends_on("python@3.7:3", when="@1.2.0:", type=("build", "run"))
     depends_on("py-poetry-core@1.0.7:1.0", when="@:1.1", type=("build", "run"))
     depends_on("py-poetry-core@1.1.0", when="@1.2.0", type=("build", "run"))
-    depends_on("py-poetry-core@1.2.0", when="@1.2.1:", type=("build", "run"))
+    depends_on("py-poetry-core@1.2.0", when="@1.2.1", type=("build", "run"))
+    depends_on("py-poetry-core@1.7.0", when="@1.6.1", type=("build", "run"))
     depends_on("py-poetry-plugin-export@1.0.6:1", when="@1.2.0:", type=("build", "run"))
     depends_on("py-poetry-plugin-export@1.0.7:1", when="@1.2.1:", type=("build", "run"))
+    depends_on("py-poetry-plugin-export@1.5.0:1", when="@1.6.1", type=("build", "run"))
     depends_on(
-        "py-backports-cached-property@1.0.2:1", when="@1.2.1: ^python@:3.7", type=("build", "run")
+        "py-backports-cached-property@1.0.2:1", when="@1.2.1:^python@:3.7", type=("build", "run")
     )
-    depends_on("py-cachecontrol@0.12.9:0.12+filecache", when="@1.2.1:", type=("build", "run"))
-    depends_on(
-        "py-cachecontrol@0.12.9:0.12+filecache", when="^python@3.6:3", type=("build", "run")
-    )
-    depends_on("py-cachy@0.3.0:0.3", type=("build", "run"))
+    depends_on("py-cachecontrol@0.12.9:0.12+filecache", when="@1.2.1:1.2", type=("build", "run"))
+    depends_on("py-cachecontrol@0.13+filecache", when="@1.6.1", type=("build", "run"))
+    depends_on("py-cachy@0.3.0:0.3", when="@:1.2", type=("build", "run"))
     depends_on("py-cleo@0.8.1:0.8", when="@:1.1", type=("build", "run"))
-    depends_on("py-cleo@1", type=("build", "run"), when="@1.2.0:")
+    depends_on("py-cleo@1", when="@1.2", type=("build", "run"))
+    depends_on("py-cleo@2.0.0:2", when="@1.6.1", type=("build", "run"))
     depends_on("py-clikit@0.6.2:0.6", when="@:1.1", type=("build", "run"))
-    depends_on("py-crashtest@0.3.0:0.3", when="^python@3.6:3", type=("build", "run"))
-    depends_on("py-html5lib@1.0:1", type=("build", "run"))
+    depends_on("py-crashtest@0.3.0:0.3", when="@:1.2", type=("build", "run"))
+    depends_on("py-crashtest@0.4.1:0", when="@1.6.1", type=("build", "run"))
```

Comment:
```suggestion
    depends_on("py-crashtest@0.4.1:0.4", when="@1.6.1", type=("build", "run"))
```
for poetry versions `0.x.y` have different rules as`1.v.w` see https://python-poetry.org/docs/dependency-specification/#caret-requirements